src/vm/analysis/read_only_checker/mod.rs

```diff
@@ -190,7 +190,7 @@ impl <'a, 'b> ReadOnlyChecker <'a, 'b> {
                 };
                 res
             },
-            StxTransfer | StxBurn |
+            StxBalance | StxTransfer | StxBurn |
```

Comment:
`StxBalance` should be a read-only operation -- so it should match on the default branch (line 163).